geomesa-hbase/geomesa-hbase-datastore/src/main/scala/org/locationtech/geomesa/hbase/coprocessor/GeoMesaCoprocessor.scala

```diff
@@ -66,8 +66,9 @@ class GeoMesaCoprocessor extends GeoMesaCoprocessorService with Coprocessor with
         val scanner = env.getRegion.getScanner(scan)
         aggregator.setScanner(scanner)
         try {
-          while (aggregator.hasNextData) {
-            results.append(aggregator.aggregate())
+          val agg = aggregator.aggregate()
```

Comment:
yeah the `notFull` check could cause it to stop while `hasNextData` is still true